proto/spacemesh/api.proto

```diff
@@ -1,166 +1,70 @@
+    rpc NodeVersion(google.protobuf.Empty) returns (SimpleString);
 
-message TransactionId {
-    bytes id = 1;
-}
+    // Returns the version of the github tag or branch used to build the node
+    rpc NodeBuild(google.protobuf.Empty) returns (SimpleString);
 
-message Transaction {
-    TransactionId tx_id = 1;
-    AccountId sender = 2;
-    AccountId receiver = 3;
-    uint64 amount = 4;
-    uint64 fee = 5;
-    TxStatus status = 6;
-    uint64 layer_id = 7;
-    uint64 timestamp = 8;
-}
-
-message AccountId {
-    string address = 1;
-}
-
-message Reward {
-    uint64 layer = 1;
-    uint64 total_reward = 2;
-    uint64 layer_reward_estimate = 3;
-    // total_reward - layer_reward_estimate = fees_estimate
-    AccountId recipient = 4;
-}
+    // Returns node status
+    rpc NodeStatus(google.protobuf.Empty) returns (NodeStatus);
 
-message Block {
-    bytes id = 1;
-    repeated Transaction transactions = 2;
-}
+    // Tell the node to start syncing
+    rpc NodeSyncStart(google.protobuf.Empty) returns (google.protobuf.Empty);
 
-message LayerId {
-    bytes id = 1;
-}
+    // Gracefully shut down the node
+    rpc NodeShutdown(google.protobuf.Empty) returns (google.protobuf.Empty);
 
-message Layer {
-    LayerId id = 1;
-    repeated Block blocks = 2;
-}
+    ///////// Mesh facade
 
-message LayerEvent {
-    LayerId layer_id = 1;
-    enum Event {
-        UNDEFINED = 0;
-        APPROVED = 1;
-        CONFIRMED = 2;
-    }
-}
+    // Returns mesh genesis time as unix epoch time
+    rpc MeshGenesisTime(google.protobuf.Empty) returns (SimpleInt);
 
-message AccountEvent {
-    enum Event {
-        UNDEFINED = 0;
-        BALANCE_CHANGE = 1;
-        NONCE_CHANGE = 2;
-    }
-    uint64 value = 1;
-}
+    // Returns current layer number
+    rpc MeshCurrentLayer(google.protobuf.Empty) returns (SimpleInt);
 
-message NodeStatus {
-    uint64 peers = 1;
-    uint64 min_peers = 2;
-    uint64 max_peers = 3;
-    bool synced = 4;
-    uint64 synced_layer = 5;
-    uint64 current_layer = 6;
-    uint64 verified_layer = 7;
-}
+    // Returns current epoch number
+    rpc MeshCurrentEpoch(google.protobuf.Empty) returns (SimpleInt);
 
-/*
-   Represents a critical node error.
-   TODO: add an enum with clearly-defined error types.
- */
-message NodeError {
-    string error = 1;
-}
+    // Returns network ID
+    rpc MeshNetId(google.protobuf.Empty) returns (SimpleInt);
 
-message SyncStatus {
-  enum SyncStatus {
-      UNDEFINED = 0;
-      NEW_LAYER_VERIFIED = 1;
-      NEW_TOP_LAYER = 2;
-      SYNCED = 3;
-      SYNCING = 4;
-      NOT_SYNCED = 5;
-  }
-}
+    // Returns number of layers per epoch (a network constant)
+    rpc MeshEpochNumLayers(google.protobuf.Empty) returns (SimpleInt);
 
-service SpacemeshService {
-    // SCALAR
+    ////////// Node streaming data facade
 
-    // Test facade
+    // Sync status events
+    rpc NodeSyncStatuses(google.protobuf.Empty) returns (stream NodeSyncStatus);
```

Comment:
Can we stick to the naming convention I proposed? `NodeStreamSyncStatus` sounds more intuitive to me. I'd like to avoid plurals (that's ambiguous) - it's clear from the `Stream` that this is a stream.

Reply:
Yes, I removed all Streams from the streaming methods because I thought they are too long and it is obvious by looking at the method signature that it returns a stream of messages. I think it is good to use plurals for streaming methods to indicate that they return a stream of messages. Right now - all methods which return streams have plural names.

Reply:
> I think it is good to use plurals for streaming methods to indicate that they return a stream of messages

I disagree here :) I think it's ambiguous. But anyway we're going to refactor into different services so it should be obvious from the context (service name) which endpoints are streams.